### PR TITLE
introduce service hooks

### PR DESCRIPTION
**Pull request by ndeloof**
Introduce Service lifecycle hooks to run commands on `post_start` or `pre_stop` inside service container

### CODE REVIEW

File: types/hooks.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package types
+
+// ServiceHook is a command to exec inside container by some lifecycle events
+type ServiceHook struct {
+	Command     ShellCommand      `yaml:"command,omitempty" json:"command"`
+	User        string            `yaml:"user,omitempty" json:"user,omitempty"`
+	Privileged  bool              `yaml:"privileged,omitempty" json:"privileged,omitempty"`
+	WorkingDir  string            `yaml:"working_dir,omitempty" json:"working_dir,omitempty"`
+	Environment MappingWithEquals `yaml:"environment,omitempty" json:"environment,omitempty"`
+}
```
Comment (glours) [inline types/hooks.go:26]: aren't we missing the extensions here?
```suggestion
	Environment MappingWithEquals `yaml:"environment,omitempty" json:"environment,omitempty"`
	
	Extensions Extensions `yaml:"#extensions,inline,omitempty" json:"-"`
}
```

**Review by glours — APPROVED** [review]
Sounds good to me, I'm just wondering if we aren't just missing the `Extensions` attribut in the `ServiceHook` struct

File: types/hooks.go (hunk 1 above)
Comment (ndeloof) [reply types/hooks.go:26]: fixed by a59035ad2bf454979d724b482c403efde4d8bee9

**Comment by laurazard** [comment]
Nice!!! 🎉

**Pull request merged**